sdks/python/apache_beam/ml/gcp/video_intelligence.py

```diff
@@ -0,0 +1,107 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+A connector for sending API requests to the GCP Video Intelligence API.
+"""
+from __future__ import absolute_import
+
+from future.utils import binary_type, text_type
+from typing import Union
+
+from apache_beam import typehints
+from apache_beam.ml.gcp import video_intelligence_helper as helper
+from apache_beam.metrics import Metrics
+from apache_beam.transforms import DoFn, ParDo, PTransform
+
+__all__ = ['AnnotateVideo']
+
+
+class AnnotateVideo(PTransform):
+  """A ``PTransform`` for annotating video using the GCP Video Intelligence API
+  ref: https://cloud.google.com/video-intelligence/docs
+  """
+  def __init__(
+      self, features, video_context=None, location_id=None, metadata=None):
+    """
+      Args:
+        features: (List[``videointelligence_v1.enums.Feature``]) Required.
+          the Video Intelligence API features to detect
+        video_context: (dict, ``videointelligence_v1.types.VideoContext``)
+          Optional.
+          Additional video context and/or feature-specific parameters.
+        location_id: (str) Optional.
+          Cloud region where annotation should take place.
+          If no region is specified, a region will be determined
+          based on video file location.
+        metadata: (Sequence[Tuple[str, str]]) Optional.
+          Additional metadata that is provided to the method.
+    """
+    super(AnnotateVideo, self).__init__()
+    self.features = features
+    self.video_context = video_context
+    self.location_id = location_id
+    self.metadata = metadata
+
+  def expand(self, pvalue):
+    return pvalue | ParDo(
+        self._VideoAnnotateFn(
+            features=self.features,
+            video_context=self.video_context,
+            location_id=self.location_id,
+            metadata=self.metadata))
+
+  @typehints.with_input_types(Union[text_type, binary_type])
+  class _VideoAnnotateFn(DoFn):
+    """ A ``DoFn`` that sends every element to the GCP Video Intelligence API
```

Comment:
DoFn does not return a PCollection. Maybe we can update this to something like

A DoFn that sends each input element to the GCP Video Intelligence API service and outputs an element with the return result of the API.

Reply:
Gotcha, thanks!